Submodule cmake contains modified content
diff --git a/cmake/external/tde2e/CMakeLists.txt b/cmake/external/tde2e/CMakeLists.txt
index b9f0dd5..3a85549 100644
--- a/cmake/external/tde2e/CMakeLists.txt
+++ b/cmake/external/tde2e/CMakeLists.txt
@@ -8,6 +8,11 @@ add_library(external_tde2e INTERFACE IMPORTED GLOBAL)
 add_library(desktop-app::external_tde2e ALIAS external_tde2e)
 
 if (DESKTOP_APP_USE_PACKAGED OR LINUX)
+    if (DESKTOP_APP_USE_TD)
+        find_package(Td REQUIRED)
+        target_link_libraries(external_tde2e INTERFACE Td::tde2e)
+        return()
+    endif()
     find_package(tde2e REQUIRED)
     target_link_libraries(external_tde2e INTERFACE tde2e::tde2e)
     return()
diff --git a/cmake/variables.cmake b/cmake/variables.cmake
index 8c496c5..bb672d3 100644
--- a/cmake/variables.cmake
+++ b/cmake/variables.cmake
@@ -30,6 +30,7 @@ cmake_dependent_option(DESKTOP_APP_DISABLE_X11_INTEGRATION "Disable all code for
 option(DESKTOP_APP_LOTTIE_USE_CACHE "Use caching in lottie animations." ON)
 cmake_dependent_option(DESKTOP_APP_USE_ALLOCATION_TRACER "Use simple allocation tracer." OFF LINUX OFF)
 option(DESKTOP_APP_USE_PACKAGED_FONTS "Use preinstalled fonts instead of bundled patched ones." OFF)
+option(DESKTOP_APP_USE_TD "Find tdlib using CMake instead of tde2e." OFF)
 option(DESKTOP_APP_DISABLE_CRASH_REPORTS "Disable crash report generation." ${no_special_target})
 option(DESKTOP_APP_DISABLE_AUTOUPDATE "Disable autoupdate." ${disable_autoupdate})
 option(DESKTOP_APP_DISABLE_QT_PLUGINS "Disable bundled Qt plugins." OFF)
