From 59722484678af1f2e92fca9b73fa98fd4f80f741 Mon Sep 17 00:00:00 2001
From: duli <duli4868@gmail.com>
Date: Wed, 17 Sep 2025 20:46:16 +0800
Subject: [PATCH] Remove debug warnning in RelaseSafe build

---
 src/apprt/gtk/class/window.zig | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/src/apprt/gtk/class/window.zig b/src/apprt/gtk/class/window.zig
index df6ea647f..953d620a4 100644
--- a/src/apprt/gtk/class/window.zig
+++ b/src/apprt/gtk/class/window.zig
@@ -287,5 +286,0 @@ pub const Window = extern struct {
-        // Add our dev CSS class if we're in debug mode.
-        if (comptime build_config.is_debug) {
-            self.as(gtk.Widget).addCssClass("devel");
-        }
-
-- 
2.51.0

